03.bowling/bowling.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+def main
+  scoreboard = ARGV[0]
+  frames = split_by_frames(scoreboard)
+  final_score = calculate_score(frames)
+  output_result(final_score)
+end
+
+def spare?(frame)
+  true if frame[0] + frame[1] == 10
+end
+
+def strike?(frame)
+  true if frame[0] == 10
+end
+
+def change_last_frame(frames, last_frame)
+  (last_frame + frames[10]).delete_if(&:zero?)
+end
+
+def split_by_frames(scoreboard)
+  scores = scoreboard.split(',')
+
+  shots = scores.flat_map do |score|
+    score == 'X' ? [10, 0] : score.to_i
+  end
+
+  frames = shots.each_slice(2).to_a
+  last_frame = frames[9]
+
+  # 最後のフレームがスペアかストライクの場合、この後のcalculate_scoreで扱いやすいように最後のフレームを3shotsにまとめる
+  frames = frames[0..-3].push(change_last_frame(frames, last_frame)) if spare?(last_frame)
+  frames = frames[0..-3].push(change_last_frame(frames, last_frame)) if strike?(last_frame) && frames[10]
+  frames
```

Comment:
`<<`と`=`を同時に使ってわかりにくかったので、`push`を代用しました。
しかし、現状ここで再代入をしないと期待している結果が得られないのが気掛かりです😕
より良い方法があればご教示いただきたいです🙇
```suggestion
  frames = frames[0..-3].push(change_last_frame(frames, last_frame)) if spare?(last_frame)
  frames = frames[0..-3].push(change_last_frame(frames, last_frame)) if strike?(last_frame) && frames[10]
  frames
```

Reply:
上にも書いたとおり、`=`は不要じゃないかなと思ってます。
もしうまく動かなければ詳しい状況を教えてください。